docs/archive-data.md

```diff
@@ -4,99 +4,75 @@ title: How to archive data
+1. **Configuration**: Archival is controlled by the [server configuration](https://github.com/temporalio/temporal/blob/master/config/development.yaml#L81) (i.e. the `config/development.yaml` file).
+2. **Provider**: Location where the data should be archived. Supported providers are S3, GCloud, and the local file system.
+3. **Archiver**: System component which archives and retrieves data. Each archiver supports a specific provider. 
+4. **URI**: Specifies which provider and archiver should be used. The system uses the URI schema and path to make the determination.
 
-The local filesystem is used mainly for local installations, testing, and when running Temporal via `docker-compose`. In a production environment, you will most likely want to archive data using one of the cloud based storage systems.
+To setup Archival, you will need to take the following steps.
 
-Get started with Google Cloud storage here: https://cloud.google.com/storage
-Get started with S3 storage here: https://aws.amazon.com/s3
+1. Setup the provider of your choice. 
+2. Configure Archival.
+3. Create a Namespace that uses a valid URI and has Archival enabled.
 
-You will need the URI of the storage location where you want the data to be archived.
+### Providers
 
-If you want to create your own archiver, review the [Custom archiver](#custom-archiver) section below. Contributions are welcome.
+Temporal supports several providers out of the box.
 
-### Archival configuration
+- **Local file system**: The [filestore archiver](https://github.com/temporalio/temporal/tree/master/common/archiver/filestore) is used to archive data in the file system of whatever host the Temporal server is running on. This provider is used mainly for local installations and testing and should not be relied on for production environments.
+- **Google Cloud**: The [gcloud archiver](https://github.com/temporalio/temporal/tree/master/common/archiver/gcloud) is used to connect and archive data with [Google Cloud](https://cloud.google.com/storage).
+- **S3**: The [s3store archiver](https://github.com/temporalio/temporal/tree/master/common/archiver/s3store) is used to connect and archive data with [S3](https://aws.amazon.com/s3). 
+- **Custom**: If you want to use a provider that is not currently supported, you can [create your own archiver](#custom-archiver) to support it.
 
-First familiarize yourself with the configuration in [`config/development.yaml`](https://github.com/temporalio/temporal/blob/master/config/development.yaml#L81).
+When you choose a provider, make sure to save the exact storage location's URI, as it is passed as a parameter when you [create a Namespace](#namespace-creation).
```

Comment:
I feel the wording could be slightly better here.

Reply:
🤷